go/vt/vttablet/tabletserver/connpool/dbconn.go

```diff
@@ -306,6 +311,23 @@ func (dbc *DBConn) Kill(reason string, elapsed time.Duration) error {
 	return nil
 }
 
+// this method is used in Kill when there is no transaction pool
+func (dbc *DBConn) killNoPool(reason string, elapsed time.Duration) error {
+	killConn, err := dbconnpool.NewDBConnection(dbc.info, tabletenv.MySQLStats)
```

Comment:
I don't think we can rely on the appdebug user's mysql credentials being sufficient to kill the query.

Instead I think we should either keep a pointer to the dbaPool around even when the pool is nil, or possibly use a different flag to tell the difference between pooled and appDebug connections.